recipe/meta.yaml

```diff
@@ -1,9 +1,8 @@
-{% set name = "cuda-version" %}
-{% set version = "12.0.0" %}
```

Comment:
Is dropping the update version intentional? So we would allow pinning to 12.0.1 but not 12.0.0 easily?

Reply:
I think this is a good question. Historically if we compare to `cudatoolkit`, it looks like we used the "update" version component but may have skipped some and may not have shipped all of the "update" versions.

https://anaconda.org/conda-forge/cudatoolkit/files

Reply:
For reference, the existing `cuda-version` package is labeled `12.0.0`.

Past versions of `cudatoolkit` are labeled as below. Copied here for ease of use in `conda_build_config.yaml`. (Note: ChatGPT helped with extracting the version numbers from some raw HTML that I copied, and sorted them correctly. I verified it for correctness.)

```yaml
- 9.2.148
- 10.0.130
- 10.1.243
- 10.2.89
- 11.0.3
- 11.1.1
- 11.2.0
- 11.2.1
- 11.2.2
- 11.3.1
- 11.4.2
- 11.5.0
- 11.5.1
- 11.6.0
- 11.7.0
- 11.8.0
```

Reply:
Was assuming users would always want patch fixes and would be more concerned about controlling major-minor versions

Are there cases when users would want to pin the patch version?

Reply:
Ideally users would be able to take the latest `12.0` without issue. The difference between `cuda-version`s `X.Y.0` and `X.Y.1` could be a single library component version bump, or the full CTK rebuild. So I could see a case for users not wanting to take on that drift and pin to `X.Y.0`.

Reply:
Ah, right. One issue with making `cuda-version` have a patch version is that updates may only be for a few packages. Currently we would have to rebuild new versions of all packages, bumping the jinja variable `cuda_version = "12.0.0"` to `cuda_version = "12.0.1"` even for packages that are unchanged in the new update. If we only released updates to the changed packages, the `cuda-version` constraint would fail to find the rest of the CTK that is still on `cuda-version==12.0.0`.

Reply:
Yeah admittedly there is a subjective element of this decision, it depends on how much work we want to take on and how valuable we think the end product of that work is

Reply:
So how should we move forward here? Adopt the version list that Bradley-ChatGPT generated and merge, but defer the pinning of major.minor.patch to a later stage?

Reply:
Given it sounds like we want to leave patch version to a higher level package (like `cuda` or `cuda-toolkit`), it sounds like we don't have a need to include the patch version in this package. Though please feel free to correct me if I've misunderstood something

Reply:
I think @jakirkham is right. We may only need to produce packages with major.minor versions.

Reply:
Sorry @jakirkham @bdice, but I still don't understand. If we don't generate `cuda-version` with the patch versions included, how do we pin in a high-level package later?

Reply:
The answer from @adibbley is that the high-level / meta- packages will have an explicit list of CUDA components (with exact versions), so we don't need to add the patch version here. I approved the PR.